packages/cli/lib/commands/create.js

```diff
@@ -119,10 +95,56 @@ function requestParams(argv) {
 	];
 }
 
+async function fetchTemplates() {
+	let templates = [];
+
+	try {
+		// fetch the repos list from the github API
+		info('Fetching official templates:\n');
+
+		// If cache file doesn't exist, then hit the API and fetch the data
+		if (!fs.existsSync(TEMPLATES_CACHE_FILENAME)) {
+			const repos = await fetch(TEMPLATES_REPO_URL).then(r => r.json());
+			await fs.writeFile(
+				TEMPLATES_CACHE_FILENAME,
+				JSON.stringify(repos, null, 2),
+				'utf-8'
+			);
+		}
+
+		// update the cache file without blocking the rest of the tasks.
+		fetch(TEMPLATES_REPO_URL)
```

Comment:
Why a mix of .then and async in same function?

Reply:
If we do `async` here then it will delay the initial bootup since it will stop the execution of the main thread.

The 2nd fetch is only for updating the cache file every time user runs `preact create`

Reply:
Will move the caching logic to a separate function.